recipes/complex-filter.md

```diff
@@ -0,0 +1,106 @@
+---
+description: I need a way to handle calls that contains many parameters in URL
+complexity: 2
+---
+
+# Complex filter
+
+## Use-case
+
+On any APIs where there is a need to fetch resources, there is also a need to filter these using search criteria.
+In most cases, these criteria are present in the url as query params. But when there is too many search criteria used to filter a list, the API call may look like this:
+
+```shell
+GET https://api.example.com/v1/products?page=1&limit=5&typeProduct=type1&typeProduct=type2&typeProduct=type3&brand=brand1&brand=brand2&...
+```
+
```

Comment:
It can be tempting to shorten the request by using shortens or code to describe your search criterias, ending up with something like 
```
GET https://api.example.com/v1/products?p=1&lim=5&tp=t1&tp=t2&tp=t3&brand=b1&brand=b2...
```
In this case, the request is even less readable, you will need a very strong documentation and you may end up developping dictionnaries to list all those codes.